nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/AdminMain.java

```diff
@@ -60,49 +78,55 @@
 import org.glassfish.api.admin.ParameterMap;
 import org.glassfish.common.util.admin.AsadminInput;
 
-import com.sun.enterprise.universal.glassfish.ASenvPropertyReader;
-import com.sun.enterprise.universal.i18n.LocalStringsImpl;
-import com.sun.enterprise.universal.io.SmartFile;
-import com.sun.enterprise.util.JDK;
-import com.sun.enterprise.util.SystemPropertyConstants;
-import java.io.File;
-import java.security.AccessController;
-import java.security.PrivilegedAction;
-import java.util.HashSet;
-import java.util.Set;
+import static fish.payara.logging.jul.PayaraLogManager.getLogManager;
+import static java.util.logging.Level.FINE;
+import static java.util.logging.Level.FINER;
+import static java.util.logging.Level.FINEST;
 
 /**
- * The admin main program (nadmin).
+ * The admin main program (asadmin).
+ * <p>
+ * If you need tracing of what it does, set AS_TRACE to true
  */
 public class AdminMain {
 
+    private static final SortedProperties LOGGING_CFG;
+    static {
+        PayaraLoggingTracer.trace(AdminMain.class, "Preconfiguring logging for asadmin.");
```

Comment:
This part was and is very ugly, because each classloader behaves in it's special way and each distribution have different traps. This works, but I agree it should get some love, especially MainHelper which does things I remember I had to debug two days.
It is not executed by OSGI, but OSGI has to know about it. But maybe later changes resolved my original issues, I will look at it today ...